### ISSUE TITLE
Topgrade: Remote "flathub" not found

### ISSUE THREAD

**Issue by Raindrac** [ublue-os/bluefin#1608]
### Describe the bug

My flatpaks aren't updating through topgrade. Every package fails to update with the reason "Remote "flathub" not found". For example:
```Unable to update runtime/org.gnome.baobab.Locale/x86_64/stable: Remote "flathub" not found
Unable to update runtime/org.gnome.clocks.Locale/x86_64/stable: Remote "flathub" not found
Unable to update runtime/org.gnome.font_viewer.Locale/x86_64/stable: Remote "flathub" not found
Unable to update runtime/org.gtk.Gtk3theme.adw-gtk3-dark/x86_64/3.22: Remote "flathub" not found
Unable to update runtime/org.gtk.Gtk3theme.adw-gtk3/x86_64/3.22: Remote "flathub" not found
Unable to update runtime/org.inkscape.Inkscape.Locale/x86_64/stable: Remote "flathub" not found
Unable to update runtime/org.kde.KStyle.Adwaita/x86_64/6.6: Remote "flathub" not found
```

### What did you expect to happen?

I expected topgrade to successfully check for and apply updates to flatpaks.

### Output of `rpm-ostree status`

```shell
● ostree-image-signed:docker://ghcr.io/ublue-os/bluefin-dx:latest
                   Digest: sha256:2f4c3e3cbacf8eaf4e026898ad79fd2a75b7243171ac7308dc0480453ede8f21
                  Version: 40.20240816.0 (2024-08-16T04:51:37Z)
          LayeredPackages: cryfs
            LocalPackages: Filesys-Modifications-1.0-2.x86_64
```

### Output of `groups`

```shell
raindrop wheel docker libvirt steam
```

**Comment by castrojo** [comment]
Your flatpak remote appears to be missing? Check with `flatpak remotes`

**Comment by Raindrac** [comment]
```
❯ flatpak remotes
Name    Options
flathub system
```
Here's the console output. Looks like it's there, I think?

**Comment by castrojo** [comment]
Can you replicate the error if you try to update via the flatpak command manually?

**Comment by Raindrac** [comment]
```❯ flatpak update
Looking for updates…
error: Error opening directory '/var/home/raindrop/.local/share/flatpak/app': No such file or directory
```
At first, I got this error. So I went into '/var/home/raindrop/.local/share/flatpak/` and deleted "app", which turned out to be a broken symlink.

![image](https://github.com/user-attachments/assets/94c7cf6c-ff80-4dd9-9222-3122a2e9b203)

```
❯ flatpak update
Looking for updates…
Unable to update runtime/app.drey.EarTag.Locale/x86_64/stable: Remote "flathub" not found
Unable to update runtime/app.drey.Warp.Locale/x86_64/stable: Remote "flathub" not found
Unable to update runtime/ca.desrt.dconf_editor.Locale/x86_64/stable: Remote "flathub" not found
<trimmed>
```

After deleting the broken symlink, I ran the command again and got the expected error. But 'cuz I was intrigued by the broken symlink, I booted up a new Bluefin virtual machine and ran `flatpak update` from there. It worked.

So I compared the folders and found that my flatpak folder on the virtual machine didn't contain a symlink to `app`, and interestingly, it didn't contain symlinks to `runtime` or `exports` either. 

The symlinks to `runtime` and `exports` on my system were working, but I deleted 'em anyway. Once I did, I ran `flatpak update` again and... it worked.

To narrow down the cause, I restored both folders and deleted em one-by-one. The symlink to `runtime` was the culprit.

But why were those symlinks there? I have no clue... But solved, I guess! I just wish I knew where these symlinks came from, 'cuz I didn't create them myself. Whatever the cause, I hope this doesn't affect someone else 😕

Notably, the working symlinks both have a creation date of my original installation:
`exports`: 6 June 2024 12∶16∶43 AM
`runtime`: 6 June 2024 12∶15∶01 AM (The culprit)
The broken `app` symlink was created on a different date.

According to fastfetch, my system was forged on June 6th, and midnight is around when I remember installing Bluefin, so that would mean those symlinks were created during the original installation. And for some reason, the issue didn't emerge until this week despite them being there all along. Strange?

I'm gonna close the issue 'cuz the installation onto my virtual machine didn't create the symlinks that caused this issue, so it should be solved for future users too. I suppose there's a chance that whatever caused this on my end might affect someone else, but hopefully not
